### PR TITLE
Add ComparableChecks extension

### LINKED ISSUE #1955: More `isLessThan`, `isGreaterThan`, etc. extension methods would be nice
Previously with `package:matcher` I could compare `Duration`s with `expect(duration1, lessThan(duration2))` since `Duration` provides an `operator <`.

It'd be nice if `package:checks` provided that too.  It'd probably be easiest to just provide such extension methods on `Subject<Comparable>`.

### PR DESCRIPTION
**Pull request by natebosch**
Closes #1955

Replaces the extension on `num` for `isGreaterThan`, `isLessThan`,
`isGreaterOrEqual`, and `isLessOrEqual` using the `Comparable` interface
instead of `operator >`.

### CODE REVIEW

**Review by kevmoo — APPROVED** [review]
LOVE IT!

File: pkgs/checks/lib/src/extensions/math.dart [hunk 1]
```diff
@@ -86,3 +86,47 @@ extension NumChecks on Subject<num> {
     });
   }
 }
+
+extension ComparableChecks<T> on Subject<Comparable<T>> {
```
Comment (kevmoo) [inline pkgs/checks/lib/src/extensions/math.dart:90]: Meh...it's almost worth it's only library. It's more than math. But it's not a biggy. 🤷

Comment (kevmoo) [reply pkgs/checks/lib/src/extensions/math.dart:90]: Core actually feels morea appropriate

Comment (natebosch) [reply pkgs/checks/lib/src/extensions/math.dart]: Sure, I'll leave it back in `core.dart` for now. It doesn't make a user facing difference, so we can shuffle things later if we want.

**Pull request merged**